acer-wmi-battery.c

```diff
@@ -65,6 +67,40 @@ MODULE_PARM_DESC(
 	"Turn battery health mode on (value > 0) or off (value = 0) during module "
 	"initialization (default value < 0: do not modify existing settings.)");
 
+static acpi_status get_battery_information(u32 index, u32 battery, u32 *result)
+{
+	u32 args[2] = { index, battery };
+	struct acpi_buffer output = { ACPI_ALLOCATE_BUFFER, NULL };
+	struct acpi_buffer input = { sizeof(args), args };
+	union acpi_object *obj;
+	acpi_status status;
+
+	status = wmi_evaluate_method(WMI_GUID, 0, 19, &input, &output);
+	if (ACPI_FAILURE(status))
+		return status;
+
+	obj = output.pointer;
+	if (!obj)
+		return AE_ERROR;
+
+	if (obj->type != ACPI_TYPE_BUFFER) {
+		kfree(obj);
+		return AE_ERROR;
+	}
+
+	if (obj->buffer.length != sizeof(u32)) {
+		pr_err("WMI battery information call returned buffer of unexpected length %u\n",
+		       obj->buffer.length);
+		kfree(obj);
+		return AE_ERROR;
+	}
+
+	*result = get_unaligned_le32(obj->buffer.pointer);
```

Comment:
Is it necessary to use `get_aligned_le32` here?

Reply:
The buffer returned by ACPI is not necessarily aligned, which means that we have to use  `get_aligned_le32` here.